src/lib/fields_derivers/fields_derivers.ml

```diff
@@ -0,0 +1,188 @@
+module type Deriver_basic_intf = sig
+  module Input : sig
+    type 'input_type t
+  end
+
+  module Creator : sig
+    type 'input_type t
+  end
+
+  module Output : sig
+    type 'output_type t
+  end
+
+  module Accumulator : sig
+    type 'input_type t
+  end
+
+  val init : unit -> 'input_type Accumulator.t
+
+  val add_field :
+       'f Input.t
+    -> ( [< `Read | `Set_and_create ]
+       , 'input_type
+       , 'f )
+       Fieldslib.Field.t_with_perm
+    -> 'input_type Accumulator.t
+    -> ('input_type Creator.t -> 'f) * 'input_type Accumulator.t
+
+  val finish :
+       ('input_type Creator.t -> 'input_type) * 'input_type Accumulator.t
+    -> 'input_type Output.t
+
+  val int_ : int Input.t
```

Comment:
I'm not sure that these should live in `Deriver_basic_intf`

Reply:
Let's sort this out when we meet tomorrow